### PR TITLE
[SPARK-50259][BUILD] Update Parquet to 1.14.4

### PR DESCRIPTION
**Pull request by Fokko**
### What changes were proposed in this pull request?

Bumping Apache Parquet to 1.14.4 because of a critical bug when writing a dictionary larger than 8kb. For a full overview of bugfixes, see: https://github.com/apache/parquet-java/releases/tag/apache-parquet-1.14.4

### Why are the changes needed?

A serious issue was discovered in the 1.14.x line: https://github.com/apache/parquet-java/releases/tag/apache-parquet-1.14.4-rc2

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing unit tests.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by Fokko** [comment]
Link to tests: https://github.com/Fokko/spark/actions/runs/11720553983/job/32646109185

File: pom.xml [hunk 1]
```diff
@@ -351,6 +351,17 @@
         <enabled>false</enabled>
       </snapshots>
     </repository>
+    <repository>
+      <id>parquet-release-candidate</id>
+      <name>Parquet Release Candidate</name>
+      <url>https://repository.apache.org/content/repositories/orgapacheparquet-1064</url>
```
Comment (Fokko) [inline pom.xml:357]:
```suggestion
      <url>https://repository.apache.org/content/repositories/orgapacheparquet-1065</url>
```

File: project/SparkBuild.scala [hunk 2]
```diff
@@ -281,7 +281,8 @@ object SparkBuild extends PomBuild {
       "gcs-maven-central-mirror" at "https://maven-central.storage-download.googleapis.com/maven2/",
       DefaultMavenRepository,
       Resolver.mavenLocal,
-      Resolver.file("ivyLocal", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns)
+      Resolver.file("ivyLocal", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns),
+      "parquet-release-candidate" at "https://repository.apache.org/content/repositories/orgapacheparquet-1064"
```
Comment (Fokko) [inline project/SparkBuild.scala:285]:
```suggestion
      "parquet-release-candidate" at "https://repository.apache.org/content/repositories/orgapacheparquet-1065"
```

**Review by Fokko — COMMENTED** [review]
RC2

File: pom.xml [hunk 3]
```diff
@@ -351,6 +351,17 @@
...
+      <id>parquet-release-candidate</id>
+      <name>Parquet Release Candidate</name>
+      <url>https://repository.apache.org/content/repositories/orgapacheparquet-1065</url>
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </repository>
```
Comment (Fokko) [inline pom.xml:364]:
```suggestion
```

File: project/SparkBuild.scala [hunk 4]
```diff
@@ -281,7 +281,8 @@ object SparkBuild extends PomBuild {
       "gcs-maven-central-mirror" at "https://maven-central.storage-download.googleapis.com/maven2/",
       DefaultMavenRepository,
       Resolver.mavenLocal,
-      Resolver.file("ivyLocal", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns)
+      Resolver.file("ivyLocal", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns),
+      "parquet-release-candidate" at "https://repository.apache.org/content/repositories/orgapacheparquet-1065"
```
Comment (Fokko) [inline project/SparkBuild.scala:285]:
```suggestion
```

File: project/SparkBuild.scala [hunk 5]
```diff
@@ -281,7 +281,6 @@ object SparkBuild extends PomBuild {
       "gcs-maven-central-mirror" at "https://maven-central.storage-download.googleapis.com/maven2/",
       DefaultMavenRepository,
       Resolver.mavenLocal,
```
Comment (Fokko) [inline project/SparkBuild.scala:283]:
```suggestion
      Resolver.mavenLocal,
      Resolver.file("ivyLocal", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns)
```

**Comment by dongjoon-hyun** [comment]
Thank you for updating the PR, @Fokko !

**Comment by Fokko** [comment]
@dongjoon-hyun Any time, some minor fixes, so we should be good to go: https://github.com/apache/parquet-java/releases/tag/apache-parquet-1.14.4

**Comment by dongjoon-hyun** [comment]
Please make the PR description up-to-date and convert it from `Draft` to normal PR after the CI passes.

**Review by yaooqinn — APPROVED** [review]

**Comment by Fokko** [comment]
@dongjoon-hyun Thanks for pointing out, I've updated the template and converted the draft into a normal PR. 👍

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Pull request closed without merging**